### PR TITLE
switching to jwt for sauce labs

### LINKED ISSUE #42: Sauce Labs: Switch to jwt add-on.
https://docs.travis-ci.com/user/jwt#Sauce-Labs

Will allow fork-based PRs to run Sauce Labs tests securely.

### PR DESCRIPTION
**Pull request by halkeye**
If my eyeballed yaml is right, this will fix #42

### CODE REVIEW

**Comment by halkeye** [comment]
:( i'll have to look into this

File: .travis.yml [hunk 1]
```diff
@@ -35,6 +34,8 @@ before_script:
   - nohup bash -c "node_modules/.bin/selenium-standalone start 2>&1 &"
 
 addons:
+  jwt:
+    - secure: t9yalh16COfTgMWI58MKD5cq3/4t1jiiGmRz7xmi5T4QkRkxYzATF+C60z2nXbnsF4RLflcRBgJZ0PbR/L2JxQ3yALasougJjb0EbVr8XoJQiyhnUEQpRxtoxn9oV9Fnp3/DDQduPTZ1zPzd20N7mOpYoqNMaVnMFj353fVzI3vph29/uakvR/nNCq7BNjcOll5+JRifpXFgsACWUalU+IR/R1xmrciw7KoLlHfe8Zy349X0mv3Ez2uugqsQ5CZnRI9vffNvIWXhuVU0sCoANydYS+d5vBjO5txrxZSv14enXFKBLaXkxViYmzQMHI2FDjx71KsKeCoGoH9ooSOfwPSXSrUFTRDN0nhow3Z3uqo3EjP/uJ0I+Hoe7gJWvUDw90Q1IX2WuoSxWU+fdgEewZyvQboaC6CSaF2rdvCfl058uIbcGhhkkGAmpwvJKSsytWOJjeTA3hpYPSHGlvVLvXER1bfAIC/iBf5VGBExotNrPSpG+PsqzhpUsFSmkjY5gpKElhcNW9Ag3r5IJJJL1xVu8aXO55T5VrmTn/jINEAo91nJZhHrjX7x0kPWzlvU5/JAAOjpcrOVfpvqjkpvmxev/JtPZl2cx8lHadPXYJw98hIagpQdkfojNI5C4ACSrTVNe/IoEujgykbh9egpe8HGsG+U+g6qogmFPQFUP/8=
```
Comment (BanzaiMan) [inline .travis.yml]: I think this needs to be a map, not a sequence:

```yaml:
addons:
  jwt:
    secure: t9ya…
```

Comment (halkeye) [inline .travis.yml]: @BanzaiMan I can switch it, but I don't think its the problem. JWT is working properly, its sauce_connect addon that isn't firiing.

**Comment by ryan-roemer** [comment]
@halkeye -- https://github.com/walmartlabs/little-loader/pull/44 passed in a branch with 

```yml
addons:
  jwt:
    - secure: SECRETSECRETSECRET
  sauce_connect: true
```

**Comment by halkeye** [comment]
yea, but thats your own branch thing. That kinda confirms my suspicion, somehow sauce_connect isn't whitelisted for prs properly

**Comment by ryan-roemer** [comment]
> yea, but thats your own branch thing

My intention with the branch is to confirm the _structure_ of the config is correct and can actually provide the secret in the "most privileged" case.

No ideas as to the internals / how whitelisting works, but just let me know if there are any experiments / etc. we can do for you...

**Comment by halkeye** [comment]
Yep thanks

I tried documenting it all https://github.com/travis-ci/travis-ci/issues/6569

:( I really thought we had this working, was so excited

**Comment by ryan-roemer** [comment]
@geekdave @Maciek416 -- Keep an eye on this branch. It looks like the sauce JWT thing is "almost there" but not quite yet...

**Comment by halkeye** [comment]
:D To defend it, JWT is working fine, sauce_connect addon isn't :D
but i agree

**Comment by ryan-roemer** [comment]
@halkeye -- Updated my comment 😉  Apologies for the lack of precision and thanks for all this work (and happy to help dogfood it here)

**Comment by ryan-roemer** [comment]
Rebuilding https://travis-ci.org/walmartlabs/little-loader/builds/158195834 per comment https://github.com/travis-ci/travis-ci/issues/6569#issuecomment-253588881 to see if external PR passes now.

**Comment by halkeye** [comment]
retesting!

**Comment by halkeye** [comment]
@ryan-roemer LOOOOK!

**Comment by ryan-roemer** [comment]
@halkeye -- Yep! (I actually just manually restarted the build before your newline commit). Weirdly, I _don't_ see CI status in this PR anymore...

Can you revert the extra newline and then I'll merge. Thanks for all your work!

**Comment by halkeye** [comment]
yea I'm not sure why the CI status stuff isn't working. I'm suspicious that github's webooks hare delayed.

**Comment by halkeye** [comment]
I kinda want to re-create this branch to confirm everything works

but i also know everything works from the daily motion build.

**Comment by ryan-roemer** [comment]
@halkeye -  that might actually be a good idea. I leave it up to you if you want to close this PR and just do a new one from scratch or if you want me to merge this one.

**Comment by BanzaiMan** [comment]
Sometimes Travis CI fails to post a status to commits for one reason or another. Could you try pushing another commit?

**Comment by halkeye** [comment]
Yea for sure, as soon as I get a minute on my laptop.

I don't think it's Travis fault. The last two.commits are not showing up under the pr tab on Travis either

**Comment by halkeye** [comment]
:( i'm not sure why its not picking up the changes.

**Comment by halkeye** [comment]
@BanzaiMan yea, none of the new ones are showing up at https://travis-ci.org/walmartlabs/little-loader/pull_requests

**Pull request closed without merging**